### PR TITLE
EARTH-24: adding twig_tweak to composer.

### PR DESCRIPTION
**Pull request by koppieesq**
# READY FOR REVIEW

# Summary
- Add twig_tweak contrib module to support Matson footer

# Associated Issues and/or People
- JIRA ticket EARTH-24
- https://github.com/stanford-earth/matson/pull/16
- Anyone who should be notified? @slowbot & @sherakama 

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -183,7 +183,8 @@
     "su-sws/stanford_basic": "dev-8.x-1.x",
 
     "stanford-earth/stanford_earth_profile": "dev-8.x-1.x",
-    "stanford-earth/matson": "dev-8.x-1.x"
+    "stanford-earth/matson": "dev-8.x-1.x",
+    "drupal/twig_tweak": "^1.8"
```
Comment (sherakama) [inline composer.json]: This is great, I just ask that this line go inline with the other drupal contrib modules in alphabetical order.

Thanks,

**Comment by sherakama** [comment]
@koppieesq, 

Also, can you include the configuration export to turn this module on by default. Thank you.

**Comment by koppieesq** [comment]
I've made the following changes:

- twig_tweak entry to composer.json is moved into the correct section and alphabetized
- I ran `blt local:update` and that added an entry for twig_tweak in configuration management, so that should be resolved

File: config/default/core.entity_view_display.paragraph.stanford_callout_text.default.yml [hunk 2]
```diff
@@ -10,15 +10,15 @@ dependencies:
     - link
     - text
 _core:
-  default_config_hash: dObhz6zghKOU8jZyeFcp6GRr3WmGMe9IEsW_QTfSQVQ
+  default_config_hash: 5VXdYfIN1qkdeVa_qszv2rgnp4AiOqEPcigFqfIdw7s
```
Comment (sherakama) [inline config/default/core.entity_view_display.paragraph.stanford_callout_text.default.yml]: Please remove the changes to this file. The hash should remain the same and the label was hidden intentionally.

File: config/default/core.entity_view_display.paragraph.stanford_textarea.default.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ dependencies:
     - field.field.paragraph.stanford_textarea.field_p_textarea
     - paragraphs.paragraphs_type.stanford_textarea
 _core:
-  default_config_hash: AVOFgnWtfSW3m3aPKydOW3VJZQ6CsXufW4ZfxEP7ZUU
+  default_config_hash: LblDvtewjUDbmm1Ja0GhP4C26s0YpbaS7bvzjBVJCV4
```
Comment (sherakama) [inline config/default/core.entity_view_display.paragraph.stanford_textarea.default.yml]: Please remove the changes to this file. The hash should remain the same and the label was hidden intentionally.

File: config/default/core.entity_view_display.paragraph.stanford_wysiwyg.default.yml [hunk 4]
```diff
@@ -8,15 +8,15 @@ dependencies:
   module:
     - text
 _core:
-  default_config_hash: zRjBJZ7_YPdGkGcBsmyyiYhH1z1U7zCiNKRg_--srzE
```
Comment (sherakama) [inline config/default/core.entity_view_display.paragraph.stanford_wysiwyg.default.yml]: Please remove the changes to this file. The hash should remain the same and the label was hidden intentionally.

File: config/default/core.extension.yml [hunk 5]
```diff
@@ -107,12 +107,7 @@ module:
   stanford_image_styles: 0
   stanford_layouts: 0
   stanford_page: 0
-  stanford_paragraph_buttons: 0
```
Comment (sherakama) [inline config/default/core.extension.yml]: These are new modules that should be enable. Please restore them.

File: config/default/field.field.node.complex_page.field_component.yml [hunk 6]
```diff
@@ -22,18 +26,22 @@ default_value_callback: ''
 settings:
   handler: 'default:paragraph'
   handler_settings:
-    target_bundles: null
```
Comment (sherakama) [inline config/default/field.field.node.complex_page.field_component.yml]: Please undo the changes to this file. What was there before was correct.

File: config/default/pathauto.pattern.node_default_pattern.yml [hunk 7]
```diff
@@ -1,13 +0,0 @@
-langcode: en
```
Comment (sherakama) [inline config/default/pathauto.pattern.node_default_pattern.yml]: Please restore this file.

File: config/default/pathauto.pattern.user_pattern.yml [hunk 8]
```diff
@@ -1,13 +0,0 @@
-langcode: en
```
Comment (sherakama) [inline config/default/pathauto.pattern.user_pattern.yml]: Please restore this file.

File: config/default/system.site.yml [hunk 9]
```diff
@@ -1,11 +1,11 @@
 uuid: ad930f39-8d32-4f67-825a-b684b189ebc4
-name: 'Drupal 8 Earth'
```
Comment (sherakama) [inline config/default/system.site.yml]: Please undo the changes to this file. The previous settings were correct.

File: config/default/user.role.anonymous.yml [hunk 10]
```diff
@@ -11,5 +11,6 @@ is_admin: false
 permissions:
   - 'access comments'
   - 'access content'
+  - 'access patterns page'
```
Comment (sherakama) [inline config/default/user.role.anonymous.yml]: I'm not sure about allowing anonymous access to the patterns page. Please remove this.

File: config/default/user.role.authenticated.yml [hunk 11]
```diff
@@ -11,6 +11,7 @@ is_admin: false
 permissions:
   - 'access comments'
   - 'access content'
+  - 'access patterns page'
```
Comment (sherakama) [inline config/default/user.role.authenticated.yml]: I think we would want to be more granular about our access to this page. Please remove this permission.

File: config/local/system.site.yml [hunk 12]
```diff
@@ -1,14 +0,0 @@
-uuid: ad930f39-8d32-4f67-825a-b684b189ebc4
```
Comment (sherakama) [inline config/local/system.site.yml]: Please restore this file.

File: config/test/system.site.yml [hunk 13]
```diff
@@ -1,14 +0,0 @@
-uuid: ad930f39-8d32-4f67-825a-b684b189ebc4
```
Comment (sherakama) [inline config/test/system.site.yml]: Please restore this file.

File: composer.json [hunk 14]
```diff
@@ -160,6 +160,7 @@
     "drupal/taxonomy_menu": "~3.3",
     "drupal/token": "~1.0",
     "drupal/token_filter": "~1.0",
+    "drupal/twig_tweak": "^1.8",
```
Comment (sherakama) [inline composer.json]: Thank you. This is great.

**Comment by sherakama** [comment]
@koppieesq,

It looks like you may have been caught in a workflow bug. The `blt local:update` command should have been the right one for you. It should pull down the latest database and apply the configuration from `/config/default` and `/config/local`. Perhaps a `git pull` or a `git merge` from the develop branch was needed before running the local update? Also, there is a bug where the system.site.yml file gets trashed on export and you need to `git checkout` to get them back again. 

Please have a look at the inline comments for changes. Thanks.

**Comment by koppieesq** [comment]
Thanks @sherakama - that explains it.  I'll redo and resubmit.

**Comment by koppieesq** [comment]
I reverted my changes, re-pulled from develop branch, and re-ran `blt local:update`, and got the following error:
```exception 'Drupal\Core\Config\ConfigImporterException' with message 'There were errors validating the config synchronization.' in /Library/WebServer/Documents/se3/docroot/core/lib/Drupal/Core/Config/ConfigImporter.php:728```
Then I ran `drush cim -y` and that seemed to work.

I've redone the following changes:

- Add twig_tweak to composer and make sure it's in the correct place in composer.json
- When I ran `composer require`, it automatically updated Drupal from 8.3.2 to 8.3.3.  Looks like it also updated some other files like `blt/project.yml`.  I hope that's ok.
- I added twig_tweak to config management so it should be on by default when you run `drush cim`

Not sure if this is a cleaner PR now - @sherakama please let me know what you think.

**Comment by sherakama** [comment]
There are a number of changes in this branch we don't want. Not sure why you deleted the drush policy and removed the comments on the project.yml file. Anyhow. I poached out what we need and merged it in to develop.

**Comment by koppieesq** [comment]
Gah.  Totally not intentional.  `drush cim` & `drush cex` didn't work as expected either - I had to finesse the results a lot before even pushing it up.

Thanks for poaching what you needed.  Hopefully this weirdness won't happen next time.
